### PR TITLE
HBASE-26419 Tracing Spans of RPCs should follow otel semantic conventions

### PR DESCRIPTION
**Pull request by ndimiduk**

### CODE REVIEW

**Comment by ndimiduk** [comment]
Checkpointing a draft PR. In summary, this patch implements connection attributes and most table operations for both `HTable` and `RawAsyncTableImpl`. Notably, actions implemented directly as `AsyncProcessTask` are not yet implemented.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 39s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m  2s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 50s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 34s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   1m 10s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 31s |  hbase-client in the patch failed.  |
| -1 :x: |  compile  |   0m 46s |  hbase-server in the patch failed.  |
| -0 :warning: |  javac  |   0m 31s |  hbase-client in the patch failed.  |
| -0 :warning: |  javac  |   0m 46s |  hbase-server in the patch failed.  |
| -1 :x: |  shadedjars  |   3m 27s |  patch has 16 errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 33s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 33s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  |   0m 31s |  hbase-client in the patch failed.  |
| -1 :x: |  unit  |   0m 46s |  hbase-server in the patch failed.  |
|  |   |  29m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3848 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7552d85add0e 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 98a4fb8086 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-compile-hbase-client.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-compile-hbase-server.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-compile-hbase-client.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-compile-hbase-server.txt |
| shadedjars | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-shadedjars.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-client.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/testReport/ |
| Max. process+thread count | 414 (vs. ulimit of 12500) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 22s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 16s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   5m 15s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   2m 20s |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   4m 18s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   1m  5s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 54s |  hbase-client in the patch failed.  |
| -1 :x: |  compile  |   0m 43s |  hbase-server in the patch failed.  |
| -0 :warning: |  javac  |   0m 54s |  hbase-client in the patch failed.  |
| -0 :warning: |  javac  |   0m 43s |  hbase-server in the patch failed.  |
| -0 :warning: |  checkstyle  |   0m 32s |  hbase-client: The patch generated 14 new + 40 unchanged - 1 fixed = 54 total (was 41)  |
| -0 :warning: |  checkstyle  |   1m 15s |  hbase-server: The patch generated 3 new + 160 unchanged - 18 fixed = 163 total (was 178)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  hadoopcheck  |   1m 38s |  The patch causes 16 errors with Hadoop v3.1.2.  |
| -1 :x: |  hadoopcheck  |   3m 14s |  The patch causes 16 errors with Hadoop v3.2.1.  |
| -1 :x: |  spotbugs  |   0m 18s |  hbase-client in the patch failed.  |
| -1 :x: |  spotbugs  |   0m 39s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   |  31m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3848 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4e6a29387b7f 4.15.0-143-generic #147-Ubuntu SMP Wed Apr 14 16:10:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 98a4fb8086 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-compile-hbase-client.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-compile-hbase-server.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-compile-hbase-client.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-compile-hbase-server.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| hadoopcheck | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-javac-3.1.2.txt |
| hadoopcheck | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-javac-3.2.1.txt |
| spotbugs | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-spotbugs-hbase-client.txt |
| spotbugs | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-general-check/output/patch-spotbugs-hbase-server.txt |
| Max. process+thread count | 86 (vs. ulimit of 12500) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  7s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 13s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   8m  8s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 35s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   1m 23s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 32s |  hbase-client in the patch failed.  |
| -1 :x: |  compile  |   0m 50s |  hbase-server in the patch failed.  |
| -0 :warning: |  javac  |   0m 32s |  hbase-client in the patch failed.  |
| -0 :warning: |  javac  |   0m 50s |  hbase-server in the patch failed.  |
| -1 :x: |  shadedjars  |   4m  2s |  patch has 16 errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 34s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 55s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  |   0m 32s |  hbase-client in the patch failed.  |
| -1 :x: |  unit  |   0m 48s |  hbase-server in the patch failed.  |
|  |   |  32m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3848 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8cd3afe37291 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 98a4fb8086 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| mvninstall | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-hbase-client.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-hbase-server.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-hbase-client.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-hbase-server.txt |
| shadedjars | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-shadedjars.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-client.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/testReport/ |
| Max. process+thread count | 287 (vs. ulimit of 12500) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3848/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — COMMENTED** [review]
Agree that this could be split into several parts...

First introduce the new SemanticAtteribute class for exmaple...

File: hbase-common/src/main/java/org/apache/hadoop/hbase/trace/SemanticAttributes.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import io.opentelemetry.api.common.AttributeKey;
+import java.util.List;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * The constants in this class correspond with the guidance outlined by the OpenTelemetry
+ * <a href="https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions">Semantic Conventions</a>.
+ */
+@InterfaceAudience.Private
+public final class SemanticAttributes {
```
[the remark this answers is not in the document]
Comment (ndimiduk) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/trace/SemanticAttributes.java:29]: The classes are in different packages, so having the same names is fine, in my opinion. It should be only in the o.a.h.h.trace.SemanticAttributes class where both classes are imported.

[the remark this answers is not in the document]
Comment (Apache9) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/trace/SemanticAttributes.java:29]: So the assumption here is we should not use the SemanticAttributes in otel? If we can make sure this then maybe using the same name is fine. Better add a ban import rule in the enforcer plugin.

**Pull request closed without merging**